### PR TITLE
Fix template preview not showing in firefox

### PR DESCRIPTION
**Pull request by DimalChandrasiri**
## Purpose
Fixes https://github.com/wso2/product-is/issues/10417

## Goals
To fix the previewing of the user entered email template content

## Approach
Added a timeout to wait till the `iframe` initialise to load the content.

### CODE REVIEW

**Review by thivi — APPROVED** [review]

File: apps/console/src/features/email-templates/components/email-template-editor.tsx [hunk 1]
```diff
@@ -82,10 +82,16 @@ export const EmailTemplateEditor: FunctionComponent<EmailTemplateEditorPropsInte
...
+        // This interval will wait till the iframe is initialized to show content.
+        const inter = window.setInterval(function() {
+            const iframeDoc = iframe?.current?.contentDocument || iframe?.current?.contentWindow?.document;
+
+            if(iframeDoc.readyState == "complete") {
+
+                window.clearInterval(inter);
+                iframeDoc.body.innerHTML = updatedContent.current;
+            }
+        },100);
```
Comment (jeradrutnam) [inline apps/console/src/features/email-templates/components/email-template-editor.tsx:94]:
```suggestion
        }, 100);
```

File: apps/console/src/features/email-templates/components/email-template-editor.tsx [hunk 2]
```diff
@@ -82,10 +82,16 @@ export const EmailTemplateEditor: FunctionComponent<EmailTemplateEditorPropsInte
...
     const writeToIframe = (): void => {
-        const iframeDoc = iframe?.current?.contentDocument || iframe?.current?.contentWindow?.document;
-        if (iframeDoc) {
-            iframeDoc.body.innerHTML = content;
-        }
+        // This interval will wait till the iframe is initialized to show content.
+        const inter = window.setInterval(function() {
+            const iframeDoc = iframe?.current?.contentDocument || iframe?.current?.contentWindow?.document;
+
+            if(iframeDoc.readyState == "complete") {
```
Comment (jeradrutnam) [inline apps/console/src/features/email-templates/components/email-template-editor.tsx:89]:
```suggestion
            if (iframeDoc.readyState === "complete") {
```

**Comment by jeradrutnam** [comment]
Why are we using an interval here? We can't depends on a static interval right?

File: apps/console/src/features/email-templates/components/email-template-editor.tsx [hunk 3]
```diff
@@ -82,10 +82,16 @@ export const EmailTemplateEditor: FunctionComponent<EmailTemplateEditorPropsInte
...
+
+                window.clearInterval(inter);
```
Comment (jeradrutnam) [inline apps/console/src/features/email-templates/components/email-template-editor.tsx:91]: E.g. Use complete word when declaring variables, unless its a commonly using term.

```suggestion
                window.clearInterval(interval);
```

Comment (DimalChandrasiri) [reply apps/console/src/features/email-templates/components/email-template-editor.tsx]: Fixed with b25631a

File: apps/console/src/features/email-templates/components/email-template-editor.tsx (hunk 2 above)
Comment (DimalChandrasiri) [reply apps/console/src/features/email-templates/components/email-template-editor.tsx]: Fixed with b25631a

File: apps/console/src/features/email-templates/components/email-template-editor.tsx (hunk 1 above)
Comment (DimalChandrasiri) [reply apps/console/src/features/email-templates/components/email-template-editor.tsx]: Fixed with b25631a

**Comment by DimalChandrasiri** [comment]
> Why are we using an interval here? We can't depends on a static interval right? Need to discuss the approach.

The implementation was changed and the `setInterval` method was replaced to a better approach. In the current approach, before setting content to the iframe, an iframe `load` event is triggered using the following method. 

```
iframeDoc.open();
iframeDoc.close(); 
```

The content is not loaded in firefox is due to it not recognising when the initial content is not set. This stackoverflow [question](https://stackoverflow.com/questions/7828502/cannot-set-document-body-innerhtml-of-iframe-in-firefox) contains the above implementation as a solution.

**Review by jeradrutnam — APPROVED** [review]

**Pull request merged**